server/src/main/kotlin/core/action/bettinground/Check.kt

```diff
@@ -0,0 +1,24 @@
+package core.action.bettinground
+
+import core.gameflow.HandState
+
+class Check : BettingAction(ActionType.CHECK) {
+
+    override fun innerApply(handState: HandState): HandState {
+        val updatedPlayer = handState.activePlayer!!.copy(lastAction = ActionType.CHECK)
+        return handState.updateActivePlayer(updatedPlayer)
+    }
+
+    override fun innerValidate(handState: HandState): ActionValidation {
+        val activePlayer = handState.activePlayer!!
+
+        return when {
+            (handState.totalBet == 0) -> ValidAction()
```

Comment:
Maybe we should also a 'lastAgressor == null' condition?

Reply:
It would be redundant imo.

Reply:
better safe than sorry

Reply:
I don't think trying to cover scenarios that shouldn't ever happen in the game is a great idea. It may make the code actually more bug-prone.